pkg/ingester/instance.go

```diff
@@ -249,33 +249,64 @@ func (i *instance) Series(_ context.Context, req *logproto.SeriesRequest) (*logp
 		return nil, err
 	}
 
-	dedupedSeries := make(map[uint64]logproto.SeriesIdentifier)
-	for _, matchers := range groups {
-		err = i.forMatchingStreams(matchers, func(stream *stream) error {
-			// exit early when this stream was added by an earlier group
-			key := stream.labels.Hash()
-			if _, found := dedupedSeries[key]; found {
-				return nil
-			}
+	var series []logproto.SeriesIdentifier
 
-			dedupedSeries[key] = logproto.SeriesIdentifier{
+	// If no matchers were supplied we include all streams.
+	if len(groups) == 0 {
+		series = make([]logproto.SeriesIdentifier, 0, len(i.streams))
```

Comment:
This allocation isn't protected by a mutex so it could be wrong but I think this would generally be ok, worst case a stream is added between the creation of this array and it's population causing Go to have to double it's size.  I'm thinking this is ok but am open to alternatives or different opinions.

Reply:
No problem here from my point of view. But if you're not sure you can write a test that does 2 query in different goroutine and run that test using -race.

`go test -mod=vendor -race  ./pkg/ingester`.

Reply:
I think you could have kept the same function `func(stream *stream) error` for both cases.`